src/client/sandbox/event/focus-blur.ts

```diff
@@ -89,7 +89,7 @@ export default class FocusBlurSandbox extends SandboxBase {
     }
 
     _shouldUseLabelHtmlForElement (el: HTMLLabelElement, type: string): boolean {
-        return type === 'focus' && domUtils.isLabelElement(el) && !!el.htmlFor;
+        return type === 'focus' && !domUtils.isElementFocusable(el) && domUtils.isLabelElement(el) && !!el.htmlFor;
```

Comment:
`domUtils.isElementFocusable(el) ` the most heavy check.
Change statements order as follows:
```js
return type === 'focus' && !!el.htmlFor  && domUtils.isLabelElement(el) && !domUtils.isElementFocusable(el) ;
```